pandas/io/formats/style.py

```diff
@@ -464,13 +472,26 @@ def _update_ctx(self, attrs: DataFrame) -> None:
             Whitespace shouldn't matter and the final trailing ';' shouldn't
             matter.
         """
+        err = KeyError(
+            "`Styler.apply` and `.applymap` are not compatible with subset slices "
+            "containing non-unique index or column keys."
+        )
+
         for cn in attrs.columns:
-            for rn, c in attrs[[cn]].itertuples():
-                if not c:
-                    continue
-                css_list = maybe_convert_css_to_tuples(c)
-                i, j = self.index.get_loc(rn), self.columns.get_loc(cn)
-                self.ctx[(i, j)].extend(css_list)
+            try:
+                for rn, c in attrs[[cn]].itertuples():
+                    if not c:
```

Comment:
can you limit the try/except not to the entire loop here, e.g. just put it around the .get_loc

Reply:
it would be even better to simply refuse to render non-uniques entirely.

Reply:
i removed the separate cases, as suggested.